tests/Paramore.Brighter.AWS.Tests/TestDoubles/MyLargeCommandMessageMapper.cs

```diff
@@ -21,3 +22,21 @@ public MyLargeCommand MapToRequest(Message message)
         return JsonSerializer.Deserialize<MyLargeCommand>(message.Body.Value);
     }
 }
+
+public class MyLargeCommandMessageMapperAsync : IAmAMessageMapperAsync<MyLargeCommand>
+{
+    [ClaimCheck(0, thresholdInKb: 5)]
+    public Task<Message> MapToMessage(MyLargeCommand request)
+    {
+        return Task.FromResult(new Message(
+            new MessageHeader(request.Id, "transform.event", MessageType.MT_COMMAND, DateTime.UtcNow),
+            new MessageBody(JsonSerializer.Serialize(request, new JsonSerializerOptions(JsonSerializerDefaults.General))))
+        );
+    }
+
+    [RetrieveClaim(0, retain:false)]
+    public Task<MyLargeCommand> MapToRequest(Message message)
+    {
+        return Task.FromResult(JsonSerializer.Deserialize<MyLargeCommand>(message.Body.Bytes));
```

Comment:
Not sure if there is a perf advantage, considering a large message, to using async deserialize here i.e. create a memory stream from body bytes, async deserialize from stream, return MyLargeCommand

Reply:
I can do that, I was testing that this would fix the issue. I can't run the aws tests local atm

Reply:
There should be AWS login information in the tests for when they run on GitHub. But it's a little harder if you don't have your kit setup for that.

Reply:
@iancooper  made the whole mapper more async-y, also realised we are missing the cancelation token.
Should the `MapToRequest(Message message) ` be `MapToRequestAsync(Message message, CancellationToken cancellationToken = default)` ?

Reply:
Good shout, we should probably keep to our naming convention here.

Reply:
I suspect that both the methods on that interface need renaming

Reply:
@iancooper I'll do it as a separate PR